sycl/test/usm/allocator_vector.cpp

```diff
@@ -30,35 +30,106 @@ int main() {
   auto dev = q.get_device();
   auto ctxt = q.get_context();
 
-  if (!dev.get_info<info::device::usm_host_allocations>()) 
+  if (!(dev.get_info<info::device::usm_host_allocations>() &&
```

Comment:
In this test you expect that a single device/environment/runtime will provide you with all three types of allocations. Dunno if it's possible that, say, CPU device/runtime provide you with either two out of three allocation facilities. Anyway, I suggest splitting the cases into three distinct ones. One way I see how this can be achieved is to `if` test blocks with appropriate `dev.get_info<info::device::usm_xxx_allocations>()`.